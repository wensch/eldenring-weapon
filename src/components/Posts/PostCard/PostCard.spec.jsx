import {render, screen} from '@testing-library/react';
import { PostCard } from "."

const mock = {
  id: '0',
  name: 'Name',
  description: 'text',
  image: 'image.png'
}

describe('<PostCard />', () => { 
  
  it('should render PostCard', () => {
    render(<PostCard {...mock} />)

    expect(screen.getByRole('img', {name: mock.name})).toHaveAttribute('src', mock.image)
    expect(screen.getByRole('heading', {name: /name/i})).toBeInTheDocument()
  })  
  
  it('should match snapshot', () => {
    const  {container}  = render(<PostCard {...mock} />)
    expect(container.firstChild).toMatchSnapshot()
  })

 })