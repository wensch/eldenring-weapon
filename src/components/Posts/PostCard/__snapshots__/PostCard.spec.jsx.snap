// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Name"
    src="image.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Name
    </h2>
    <p>
      text
    </p>
  </div>
</div>
`;
